index.js

```diff
@@ -31,24 +33,98 @@ exports.register = (server, options) => {
   // expose sentry client at server.plugins['hapi-sentry'].client
   server.expose('client', Sentry);
 
-  server.ext({
-    type: 'onRequest',
-    method(request, h) {
-      if (opts.useDomainPerRequest) {
-        // Sentry looks for current hub in active domain
-        // Therefore simply by creating&entering domain Sentry will create
-        // request scoped hub for breadcrumps and other scope metadata
-        request.__sentryDomain = domain.create();
-        request.__sentryDomain.enter();
-      }
+  // Set up request interceptor for creating and destroying a domain on each request
+  // It'll wrap the request handler returned by the _dispatch function in HAPI core
+  // allowing it to create the domain before any of HAPIs processing starts
+  // thus allowing us to use the normal HAPI extensions to add context to Sentry scopes
+  // https://github.com/hapijs/hapi/blob/c95985e225fa09c4b640a887ccb4be46dbe265bc/lib/core.js#L507-L538
+  function interceptor(next, req, res, ...args) {
+    const local = domain.create(); // Create domain to hold context for request
+    local.add(req);
+    local.add(res);
+
+    let rtn;
```

Comment:
what does this stand for? return? can we find a better less cryptic name? maybe `result` ?

Reply:
Yeah the return value of the original function that's being intercepted or wrapped. The value in our case is always `undefined` because you're not meant to return a value to event emitter. But it's good practice to send any result you might get back upstream just in case.

I'll add a comment explaining this.